GERTi/GERTiMNC.lua

```diff
@@ -146,24 +166,35 @@ end
 
 -- Used in handler["OPENROUTE"]
 -- MNC's openRoute and Client's openRoute DIFFER
-local function orController(destination, origination, beforeHop, hopOne, hopTwo, receivedPort, transmitPort, outbound)
-	print("Opening Route")
-	if modem.address ~= destination then
+local function orController(destination, origination, beforeHop, hopOne, hopTwo, receivedPort, transmitPort, cb, outbound)
+        print("Opening Route")
+        local function sendOKResponse()
+                transmitInformation(beforeHop, receivedPort, "ROUTE OPEN", destination, origination)
+		storeConnections(origination, destination, beforeHop, nextHop, receivedPort, transmitPort)
+                if cb then
+                        cb(true)
+                end
+        end
+        if modem.address ~= destination then
 		transmitInformation(hopOne, transmitPort, "OPENROUTE", destination, hopTwo, origination)
-		local eventName, receivingModem, _, port, distance, payload = event.pull(2, "modem_message")
-		if payload ~= "ROUTE OPEN" then
-			return false
-		end
-		return transmitInformation(beforeHop, receivedPort, "ROUTE OPEN"), storeConnections(origination, destination, beforeHop, hopOne, receivedPort)
-	else
-		return transmitInformation(beforeHop, receivedPort, "ROUTE OPEN"), storeConnections(origination, destination, beforeHop, hopOne, receivedPort, outbound)
-	end
+                -- Basically: When the ROUTE OPEN comes along, check if it's the route we want (it's one route per pair anyway)
+                addTempHandler(6, "ROUTE OPEN", function (eventName, recv, sender, port, distance, code, pktDest, pktOrig)
+                        if (destination == pktDest) and (origination == pktOrig) then
+                                sendOKResponse()
+                                return true -- This terminates the wait
+                        end
+                end, function () if cb then cb(false) end end)
+                return
+        end
+        sendOKResponse()
```

Comment:
Why is there a sendOKResponse() at the end? Everything should have returned already, if so, this is not needed.